src/mqt/predictor/evaluation.py

```diff
@@ -84,9 +84,11 @@ def create_tket_result(
         FullPeepholeOptimise(target_2qb_gate=OpType.TK2).apply(tket_qc)
         PlacementPass(GraphPlacement(arch)).apply(tket_qc)
         RoutingPass(arch).apply(tket_qc)
-        native_rebase.apply(tket_qc)
         duration = time.time() - start_time
         transpiled_qc_tket = tk_to_qiskit(tket_qc)
+        # replace "node" with "q" to match qiskit naming convention
+        # to prevent transpile errors in expected_success_probability(...)
+        transpiled_qc_tket.qregs[0]._name = "q"  # noqa: SLF001
```

Comment:
Is this line of code really necessary and cannot be worked-around otherwise? Modifying the private attributes of the qregs gives me a bad feeling, especially since we are currently using our own conversions between circuit formats when determining the layout attribute after a mapping pass has been applied. Although I cannot point you to an explicit line where this change here causes problems, I still have the feeling that this solution could lead to unexpected behavior.

Reply:
Agreed. However [this check](https://github.com/Qiskit/qiskit/blob/c6c45a1de9140d3051f34a90a8486e689f0a8749/qiskit/transpiler/passes/scheduling/asap.py#L59) requires us to comply with Qiskits naming convention, if we want to run the ASAP schedule.

Reply:
You should be able to achieve the same result by running some Qiskit pass on the respective circuit instead of manipulating private properties though.

Reply:
Do you mean a custom pass or just any pass that does not change the circuit?
For example, this does not change the `qreg` names:
```
pm = PassManager()
equiv_lib = EquivalenceLibrary()
pm.append(BasisTranslator(equiv_lib, device.basis_gates))
transpiled_qc_tket = pm.run(transpiled_qc_tket)
```

Reply:
I don't really have capacity to search for it, but somewhere in qiskit's compilation flow there is a pass that consolidates the registers. I think it is even called something like that

Reply:
No worries; thank you for the information, just wanted to make sure I got your suggestion right.
I'll figure it out!